### PR TITLE
Add bitshuffle page

### PR DESCRIPTION
**Pull request by kangpinghuang**
1. bitshuffle page
2. add CMakeLists.txt for segment_v2
3. unit tests

### CODE REVIEW

**Review by imay — CHANGES_REQUESTED** [review]

File: be/src/olap/rowset/segment_v2/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+# specific language governing permissions and limitations
+# under the License.
+
+# where to put generated libraries
+set(LIBRARY_OUTPUT_PATH "${BUILD_DIR}/src/olap/rowset/segment_v2")
+
+# where to put generated binaries
+set(EXECUTABLE_OUTPUT_PATH "${BUILD_DIR}/src/olap/rowset/segment_v2")
+  
+add_library(SegmentV2 STATIC
```
[the remark this answers is not in the document]
Comment (kangpinghuang) [reply be/src/olap/rowset/segment_v2/CMakeLists.txt]: Is there any possibility that SegmentV2 will be used as an independent library?

File: be/src/olap/rowset/segment_v2/bitshuffle_page.cpp [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+#include "olap/rowset/segment_v2/bitshuffle_page.h"
+#include "olap/rowset/segment_v2/common.h"
+
+namespace doris {
+namespace segment_v2 {
+
+void abort_with_bitshuffle_error(int64_t val) {
+    switch (val) {
+    case -1:
+      LOG(FATAL) << "Failed to allocate memory";
```
[the remark this answers is not in the document]
Comment (kangpinghuang) [reply be/src/olap/rowset/segment_v2/bitshuffle_page.cpp]: OK

File: be/src/olap/rowset/segment_v2/bitshuffle_page.h [hunk 3]
```diff
@@ -0,0 +1,343 @@
...
+    Slice _data;
+    bool _parsed;
+    rowid_t _page_first_ordinal;
+    size_t _num_elements;
+    size_t _compressed_size;
+    size_t _num_element_after_padding;
+
+    int _size_of_element;
+    size_t _cur_index;
+    kudu::faststring _decoded;
```
[the remark this answers is not in the document]
Comment (kangpinghuang) [reply be/src/olap/rowset/segment_v2/bitshuffle_page.h]: OK, I will change it to doris

File: be/src/olap/rowset/segment_v2/bitshuffle_page.h [hunk 4]
```diff
@@ -0,0 +1,343 @@
...
+            return Status(ss.str());
+        }
+
+        RETURN_IF_ERROR(_decode());
+        _parsed = true;
+        return Status::OK;
+    }
+
+    Status seek_to_position_in_page(size_t pos) override {
+        CHECK(_parsed) << "Must call init()";
```
[the remark this answers is not in the document]
Comment (kangpinghuang) [reply be/src/olap/rowset/segment_v2/bitshuffle_page.h]: OK

**Review by imay — APPROVED** [review]

**Pull request merged**